content/doc/tutorials/build-a-cpp-app-with-jenkins.adoc

```diff
@@ -0,0 +1,316 @@
+
+You can now select *Pipeline Overview* to see the stages of the Pipeline.
+
+[.boxshadow]
+image:tutorials/cpp/pipeline-overview-test-stage.png[alt="Pipeline overview with Test stage",width=100%]
+
+Notice the additional "Test" stage. You can select the "Test" stage checkmark to access the output from that stage.
+
+[.boxshadow]
+image:tutorials/cpp/pipeline-console-test-stage.png[alt="Test stage output",width=100%]
+
+=== Add a final delivery stage to your Pipeline
+
+. Go back to your text editor/IDE and ensure your `Jenkinsfile` is open.
+. Copy and paste the following Declarative Pipeline syntax immediately under the `Test` stage of your `Jenkinsfile`:
++
+[source,groovy]
+----
+        stage('Deliver') {
+            steps {
+                sh 'tar -czf casino_game.tar.gz build/casino_game'
+                archiveArtifacts artifacts: 'casino_game.tar.gz', fingerprint: true
+            }
+        }
+----
+so that you will end up with:
++
+[source,groovy]
+----
+pipeline {
+    agent any
+    stages {
+        stage('Build') {
+            steps {
+                sh 'rm -rf build'
+                sh 'cmake -B build -S .'
+                sh 'cmake --build build'
+            }
+        }
+        stage('Test') {
+            steps {
+                sh './build/casino_game'
+                sh './build/test_game'
+            }
+        }
+        stage('Deliver') {
+            steps {
+                sh 'tar -czf casino_game.tar.gz build/casino_game'
+                archiveArtifacts artifacts: 'casino_game.tar.gz', fingerprint: true
+            }
+        }
+    }
+}
+----
+The above defines a new stage called `Deliver` that appears on the Jenkins UI.
+This link:/doc/pipeline/steps/workflow-durable-task-step/#code-sh-code-shell-script[`sh`] step packages the executable into a `.tar.gz` file.
+This link:/doc/pipeline/steps/workflow-durable-task-step/#code-archiveartifacts-code-archive-the-artifacts[`archiveArtifacts`] step archives the `.tar.gz` file as a build artifact.
+
+. Save your edited `Jenkinsfile` and commit it to your local `casino-number-guessing-game` Git repository. Within the `casino-number-guessing-game` directory, run the commands: +
+  `git add Jenkinsfile` +
+  then +
+  `git commit -m "Add 'Deliver' stage"` +
+  and finally +
+  `git push` to push your changes to your forked repository on GitHub, so it can be picked up by Jenkins.
+
+. In Jenkins, sign in if necessary, go back to the *Dashboard* and then navigate to *casino-number-guessing-game*. Alternatively, you can go directly to *casino-number-guessing-game* depending on where you're starting from.
+. Select *Build Now* on the left. You should see after a while a new column *Deliver* appear in the *Stage View*.
+
+[.boxshadow]
+image:tutorials/cpp/deliver-overview.png[alt="Deliver stage",width=100%]
+
+. Select *#3* or on the number representing your last build on the left, under *Build History*. 
```

Comment:
```suggestion
Select *#3*, or on the number representing your last build on the left, under *Build History*. 
```
Remove numbered step as it is not part of the step-by-step instructions, add comma to include text better.